src/peripheral/mod.rs

```diff
@@ -100,13 +104,65 @@ pub struct Cpuid {
     pub isar: [RO<u32>; 5],
     reserved1: u32,
     /// Cache Level ID
+    #[cfg(armv7m)]
     pub clidr: RO<u32>,
     /// Cache Type
+    #[cfg(armv7m)]
     pub ctr: RO<u32>,
     /// Cache Size ID
+    #[cfg(armv7m)]
     pub ccsidr: RO<u32>,
     /// Cache Size Selection
-    pub csselr: RO<u32>,
+    #[cfg(armv7m)]
+    pub csselr: RW<u32>,
+}
+
+#[cfg(armv7m)]
+const CSSELR_IND_POS: u32 = 0;
+#[cfg(armv7m)]
+const CSSELR_IND_MASK: u32 = 1 << CSSELR_IND_POS;
+#[cfg(armv7m)]
+const CSSELR_LEVEL_POS: u32 = 1;
+#[cfg(armv7m)]
+const CSSELR_LEVEL_MASK: u32 = 0x7 << CSSELR_LEVEL_POS;
+#[cfg(armv7m)]
+const CCSIDR_NUMSETS_POS: u32 = 13;
+#[cfg(armv7m)]
+const CCSIDR_NUMSETS_MASK: u32 = 0x7FFF << CCSIDR_NUMSETS_POS;
+#[cfg(armv7m)]
+const CCSIDR_ASSOCIATIVITY_POS: u32 = 3;
+#[cfg(armv7m)]
+const CCSIDR_ASSOCIATIVITY_MASK: u32 = 0x3FF << CCSIDR_ASSOCIATIVITY_POS;
+
+
+impl Cpuid {
+    /// Selects the current CCSIDR
+    ///
+    /// * `level`: the required cache level minus 1, e.g. 0 for L1, 1 for L2
+    /// * `ind`: select instruction cache (1) or data/unified cache (0)
+    #[cfg(armv7m)]
```

Comment:
Sure, will do. I avoided it because "what if you wanted some other non-arm7-specific methods on CPUID"? but of course they can go in another impl block.